### PR TITLE
feat: add new user with USER_GID parameter & replace root user

### LINKED ISSUE #63: Do not run api service as root
Currently the Fast API service is running as root, it should not:

- introduce a user in backend/Dockerfile
- this user should use the USER_UID parameter to get right uid

### PR DESCRIPTION
**Pull request by Aju100**
Fixed #63

### CODE REVIEW

**Comment by Aju100** [comment]
@alexgarel Can you please review out this PR

**Review by alexgarel — COMMENTED** [review]
@Aju100 so far so good, but you also have to change docker-compose and split the build part (with args) in a docker/dev.yml as you did for facets-knowledge-panels.

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+    useradd -u $USER_UID -g off -m off && \ 
+    mkdir -p /home/off && \ 
+    chown off:off -R /code /home/off
+
 COPY ./requirements.txt /code/requirements.txt
-RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt
-COPY ./editor /code/editor
+
+USER off:off
+RUN pip3 install --no-cache-dir --upgrade -r /code/requirements.txt
```
[the remark this answers is not in the document]
Comment (alexgarel) [reply backend/Dockerfile:20]: Also as we use BUILDKIT, we can indeed use a cache and mount a cache for that.

```Dockerfile
RUN --mount=type=cache,id=pip-cache,target=/root/.cache/pip pip install --upgrade -r /code/requirements.txt 
```

**Comment by alexgarel** [comment]
@Aju100, a tip: when you fork a project, don't work on branch main, but create a new branch for you feature, then create a PR against source project.

If you work on main branch, you won't be able to resync your fork with upstream project.

**Comment by aadarsh-ram** [comment]
Thanks for the contribution @Aju100! 🎉

**Comment by Aju100** [comment]
Thank you @alexgarel for the feedback, will definitely check out it in an upcoming collaboration. Also thank you @aadarsh-ram appreciated it.

**Pull request merged**